### PR TITLE
Feature/sign up test

### PR DESCRIPTION
**Pull request by santiagoRD**
## Summary

added tests to validate signup feature

## Screenshots

tests

## Trello Card

https://trello.com/c/WsjkJiDM/103-tests-de-sign-up

### CODE REVIEW

File: src/screens/SignUp/index.tsx [hunk 1]
```diff
@@ -77,12 +83,27 @@ function SignUp() {
           errorMessage={errors.confirmPassword?.message}
         />
         {error && (
-          <p className={styles.error}>{i18next.t(`${I18N_CONFIG.key}:${ERROR_MESSAGES.signUpService}`)}</p>
+          <span role="error" className={styles.error}>
+            {i18next.t(`${I18N_CONFIG.key}:${ERROR_MESSAGES.signUpService}`)}
+          </span>
         )}
-        <button className={`${styles.submitButton} ${styles.formButton} m-bottom-5 full-width`} type="submit">
+        {state && (
```
Comment (manugo-dev) [inline src/screens/SignUp/index.tsx:94]: I'm not sure if this is the best way to handle successful response. Better way for me it's using withPostSuccess function and setting a success state and checking it

File: src/screens/SignUp/signUp.test.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { RESPONSE_STATUS } from 'constants/index';
+
+import SignUp from './index';
+
+describe('Testing SignUp Component', () => {
+  const server = setupServer();
+  rest.post(`${process.env.REACT_APP_BASE_URL}/users`, (req, res, ctx) =>
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: move this and server declaration outside the describe function.

File: src/screens/SignUp/signUp.test.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    res(ctx.status(RESPONSE_STATUS.ok), ctx.json({ ok: true }))
+  );
+  beforeEach(() => {
+    render(<SignUp />);
+  });
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: same for this. move outside.

File: src/screens/SignUp/signUp.test.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  afterAll(() => server.close());
+  test('Should validate empty fields onSubmit', async () => {
+    const allFieldsEmpty = 5;
+    userEvent.click(screen.getByRole('textbox', { name: /SignUp:email/ }));
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: Why are you testing "onSubmit" when you clicked a text box? You must click the submit button.

File: src/screens/SignUp/signUp.test.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    expect(await screen.findAllByRole('alert')).toHaveLength(allFieldsEmpty);
+  });
+
+  test('Should display error when email value is invalid', async () => {
+    userEvent.type(screen.getByLabelText(/SignUp:email/i), 'invalid-email');
+    userEvent.click(screen.getByRole('textbox', { name: /SignUp:email/ }));
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx:32]: same here

File: src/screens/SignUp/signUp.test.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    await waitFor(() => {
+      expect((screen.getByLabelText(/SignUp:email/i) as HTMLInputElement).value).toBe('invalid-email');
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: `expect(await screen.findByText(/SignUp:email/i)).toBeInTheDocument();` its another better way to do that

File: src/screens/SignUp/signUp.test.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    });
+  });
+
+  test('should can`t submit form having at least one error', async () => {
+    const onSubmit = jest.fn();
+    userEvent.type(screen.getByLabelText(/SignUp:email/i), 'invalid-email');
+    userEvent.click(screen.getByRole('textbox', { name: /SignUp:email/ }));
+    userEvent.click(screen.getByRole('button', { name: /signup/i }));
+    await waitFor(() => {
+      expect(onSubmit).not.toBeCalled();
```
Comment (manugo-dev) [inline src/screens/SignUp/signUp.test.tsx]: Did you try this function in the opposite case? when the fields are all correct? I think the onSubmit as you declared it is not mocked

File: src/screens/SignUp/signUp.test.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { RESPONSE_STATUS } from 'constants/index';
+
+import SignUp from './index';
+
```
Comment (CrisVega08) [inline src/screens/SignUp/signUp.test.tsx:11]: What do you think about this 

```
const server = setupServer(rest.post('/users', (_, res, ctx) => res(ctx.json({mockResponse}))));

beforeAll(() => server.listen());
beforeEach(() => render(<signUp>))
afterEach(() => server.resetHandlers());
afterAll(() => {
  server.close();
  cleanup();
});
```

Where mockResponse is similar to :

```
  id: xxx,
  firstName: '',
  lastName: '',
  email: '',
  locale: ''
```

File: src/screens/SignUp/signUp.test.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    userEvent.click(screen.getByRole('textbox', { name: /SignUp:email/ }));
+    userEvent.click(screen.getByRole('button', { name: /signup/i }));
+    await waitFor(() => {
+      expect(onSubmit).not.toBeCalled();
+    });
+  });
+
+  test('should call onSubmit function after click submit button with all fields full', async () => {
+    server.use(
+      rest.post(`${process.env.REACT_APP_BASE_URL}/users`, (req, res, ctx) =>
```
Comment (CrisVega08) [inline src/screens/SignUp/signUp.test.tsx]: I think environment variable is unnecessary

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**